pkg/controller/account/ec2.go

```diff
@@ -142,25 +140,37 @@ func CreateEC2Instance(reqLogger logr.Logger, client awsclient.Client, ami strin
 		}
 		// Specify the details of the instance that you want to create.
 		runResult, runErr := client.RunInstances(&ec2.RunInstancesInput{
-			// An Amazon Linux AMI ID for t2.micro instances in the us-west-2 region
+			// An Amazon Linux AMI ID for t2.micro instances in a given region:
 			ImageId:      aws.String(ami),
 			InstanceType: aws.String(awsInstanceType),
 			MinCount:     aws.Int64(1),
 			MaxCount:     aws.Int64(1),
 		})
-		if runErr == nil {
-			instanceID = *runResult.Instances[0].InstanceId
-			break
-		}
 
-	}
+		// Return on unexpected errors:
+		if runErr != nil {
+			if aerr, ok := runErr.(awserr.Error); ok {
+				switch aerr.Code() {
+				case "PendingVerification", "OptInRequired":
+					continue
+				default:
+					reqLogger.Error(runErr,
+						fmt.Sprintf(`Failed while trying to create EC2 instance, 
+							AWS Error Code: %s, 
+							AWS Error Message: %s`,
+							aerr.Code(),
+							aerr.Message()))
+					return "", runErr
+				}
+			}
+		}
 
-	if runErr != nil {
-		return "", runErr
+		// No error was found, save instance id and return
```

Comment:
I think this maybe makes more sense? "No error was found, instance is running return instance Id"